src/System.Private.ServiceModel/tools/test/SelfHostWcfService/WcfService.cs

```diff
@@ -367,6 +368,23 @@ public void ReturnContentType(string contentType)
             httpRespononseMessageProperty.Headers[HttpResponseHeader.ContentType] = contentType;
         }
 
+        public string EchoCookie()
+        {
+            string cookie = WebOperationContext.Current.IncomingRequest.Headers[HttpRequestHeader.Cookie];
+            return cookie;
+        }
+
+        public string EchoTimeAndSetCookie(string name)
+        {
+            string cookie = WebOperationContext.Current.IncomingRequest.Headers[HttpRequestHeader.Cookie];
```

Comment:
Are you using the value "cookie" here for anything? Seems like you've read it and then don't do anything else with it.